documentation/modules/exploit/linux/upnp/dlink_msearch_unauth_lan_rce.md

```diff
@@ -0,0 +1,265 @@
+- [ ] FirmAE running on Kali Linux 2023.3
+* D-Link Router model DIR-300 revisions Ax with firmware v1.06 or older;
+* D-Link Router model DIR-300 revisions Bx with firmware v2.15 or older;
+* D-Link Router model DIR-600 revisions Bx with firmware v2.18 or older;
+* D-Link Router model DIR-645 revisions Ax with firmware v1.05 or older;
+* D-Link Router model DIR-815 revisions Bx with firmware v1.04 or older;
+* D-Link Router model DIR-816L revisions Bx with firmware v2.06 or older;
+* D-Link Router model DIR-817LW revisions Ax with firmware v1.04b01_hotfix or older;
+* D-Link Router model DIR-818LW revisions Bx with firmware v2.05b03_Beta08 or older;
+* D-Link Router model DIR-822 revisions Bx with firmware v2.03b01 or older;
+* D-Link Router model DIR-822 revisions Cx with firmware v3.12b04 or older;
+* D-Link Router model DIR-823 revisions Ax with firmware v1.00b06_Beta or older;
+* D-Link Router model DIR-860L revisions Ax with firmware v1.12b05 or older;
+* D-Link Router model DIR-859 revisions Ax with firmware v1.06b01Beta01 or older;
+* D-Link Router model DIR-860L revisions Ax with firmware v1.10b04 or older;
+* D-Link Router model DIR-860L revisions Bx with firmware v2.03b03 or older;
+* D-Link Router model DIR-865L revisions Ax with firmware v1.07b01 or older;
+* D-Link Router model DIR-868L revisions Ax with firmware v1.12b04 or older;
+* D-Link Router model DIR-868L revisions Bx with firmware v2.05b02 or older;
+* D-Link Router model DIR-869 revisions Ax with firmware v1.03b02Beta02 or older;
+* D-Link Router model DIR-880L revisions Ax with firmware v1.08b04 or older;
+* D-Link Router model DIR-890L/R revisions Ax with firmware v1.11b01_Beta01 or older;
+* D-Link Router model DIR-885L/R revisions Ax with firmware v1.12b05 or older;
+* D-Link Router model DIR-895L/R revisions Ax with firmware v1.12b10 or older;
+* probably more looking at the scale of impacted devices :-(
+
+### Installation steps to emulate the router firmware with FirmAE
+* Install `FirmAE` on your Linux distribution using the installation instructions provided [here](https://github.com/pr0v3rbs/FirmAE).
+* To emulate the specific firmware that comes with the D-Link devices, `binwalk` might need to be able to handle a sasquatch filesystem.
+* Follow installation and compilation steps that you can find [here](https://gist.github.com/thanoskoutr/4ea24a443879aa7fc04e075ceba6f689).
+* Please do not forget to run this after your `FirmAE` installation otherwise you will not be able to extract the firmware.
+* Download  the vulnerable firmware from D-Link [here](http://legacyfiles.us.dlink.com/).
+* Pick `DIR-865L_REVA_FIRMWARE_1.07.B01.ZIP` for the demonstration.
+* Start emulation.
+* First run `./init.sh` to initialize and start the Postgress database.
+* Start a debug session  `./run.sh -d d-link /root/FirmAE/firmwares/DIR-865L_REVA_FIRMWARE_1.07.B01.ZIP`
+* This will take a while, but in the end you should see the following...
+
+```ShellSession
```

Comment:
I think we usually use `shell` in this scenario. More info here: https://github.com/github-linguist/linguist
```suggestion
```shell
```

Reply:
See https://github.com/rapid7/metasploit-framework/pull/18463/commits/51523e09710cc1efff0e04a367cc46481c05d913.